.github/workflows/build_test_fmt.yml

```diff
@@ -0,0 +1,59 @@
+name: cargo-gccrs CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      master
+
+jobs:
+  coding-style:
```

Comment:
With this CI setup you'll rebuild this project over and over again with every job. What you actually want to do is to have one job, that has multiple steps. 

So in the end it should something like this:

```yaml
jobs:
  build_and_test:
    runs-on: ubuntu-latest
    
    steps:
    - uses: checkout
    - name: Run rustfmt
      run: cargo fmt -- --check
    # NOTE: no need to build twice
    - name: Run Clippy
      env:
        RUSTFLAGS: "-D warnings" # Trigger an error on warnings
      run: cargo clippy
    - name: Build
      env:
        RUSTFLAGS: "-D warnings" # Trigger an error on warnings
      run: cargo build
    # NOTE: also consider combining cargo build and cargo test and setting `RUSTFLAGS` globally  
    - name: Test
      env:
        RUSTFLAGS: "-D warnings" # Trigger an error on warnings
      run: cargo test
```

I also strongly recommend to include Clippy in the CI pipeline. 

I don't think you need to check if a release build goes through. That should just work, if the debug build works.

Reply:
Gotcha, I had it setup this way so that jobs wouldn't run unless some conditions were met first. For example, no need to run the tests if the coding style isn't correct, since that's just wasting CI time. But it's also less pleasant for users who might want to rely on the CI (not that there's any need of that for such a tiny program)

Reply:
If one step in a job fails, the whole job fails. So you get the same with only one job. Just put the steps in the right order.